code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -185,3 +230,161 @@
+	X.do_jitter_animation(1000)
+
+
+	//Roar SFX; volume scales with rage
+	playsound(X.loc, 'sound/voice/alien_roar2.ogg', clamp(100 * rage_power, 25, 80), 0)
+
+	var/bonus_duration
+	if(rage_power > 0.5) //If we're super pissed it's time to get crazy
+		var/datum/action/xeno_action/charge = X.actions_by_path[/datum/action/xeno_action/activable/charge]
+		var/datum/action/xeno_action/ravage = X.actions_by_path[/datum/action/xeno_action/activable/ravage]
+		if(charge)
+			charge.clear_cooldown() //Reset charge cooldown
+		if(ravage)
+			ravage.clear_cooldown() //Reset ravage cooldown
+
+		for(var/mob/living/witness in hearers(rage_power_radius, X)) //Roar that applies cool SFX and some soft CC
+
+			if(!witness.hud_used)
+				continue
+
+			var/obj/screen/plane_master/floor/OT = witness.hud_used.plane_masters["[FLOOR_PLANE]"]
+			var/obj/screen/plane_master/game_world/GW = witness.hud_used.plane_masters["[GAME_PLANE]"]
+
+			addtimer(CALLBACK(OT, /atom.proc/remove_filter, "rage_outcry"), 1 SECONDS)
+			GW.add_filter("rage_outcry", 2, list("type" = "radial_blur", "size" = 0.07))
+			animate(GW.get_filter("rage_outcry"), size = 0.12, time = 5, loop = -1)
+			OT.add_filter("rage_outcry", 2, list("type" = "radial_blur", "size" = 0.07))
+			animate(OT.get_filter("rage_outcry"), size = 0.12, time = 5, loop = -1)
+			addtimer(CALLBACK(GW, /atom.proc/remove_filter, "rage_outcry"), 1 SECONDS)
+
+	for(var/turf/affected_tiles as() in RANGE_TURFS(rage_power_radius, X.loc))
+		affected_tiles.Shake(4, 4, 1 SECONDS) //SFX
+
+	for(var/mob/living/L in hearers(rage_power_radius, X)) //Roar that applies cool SFX and some soft CC
+		shake_camera(L, 1 SECONDS, 1)
+		L.Shake(4, 4, 1 SECONDS) //SFX
+
+		if(L.stat == DEAD) //We don't care about the dead
+			continue
+
+		if(isxeno(L))
+			var/mob/living/carbon/xenomorph/friendly_check = L
+			if(friendly_check.issamexenohive(X)) //No friendly fire
+				continue
+
+		L.adjust_stagger(rage_power_radius * 0.5) //Apply soft CC debuffs
+		L.add_slowdown(rage_power_radius * 0.5)
+		to_chat(L, "<span class='xenohighdanger'>The [X]'s terrible roar penetrates your senses, leaving you staggered.</span>")
+
+	X.add_filter("ravager_rage_outline", 5, list("type" = "outline", "size" = 1.5, "color" = COLOR_RED)) //Set our cool aura; also confirmation we have the buff
+
+	X.plasma_stored += X.xeno_caste.plasma_max * rage_power //Regain a % of our maximum plasma scaling with rage
+
+	X.soft_armor = X.soft_armor.modifyAllRatings(CEILING(rage_power * 40,1)) //Set rage armor bonus
+	switch(rage_power) //Set rage explosive resist bonus
+		if(0.25 to 0.50)
+			X.soft_armor = X.soft_armor.setRating(bomb = XENO_BOMB_RESIST_2)
+		if(0.5 to INFINITY)
+			X.soft_armor = X.soft_armor.setRating(bomb = XENO_BOMB_RESIST_3)
+
+	X.adjust_sunder(-50 * rage_power) //Restore up to 50 sunder; this armor bonus means literally nothing if we're literally shredded to shit.
+
+	X.xeno_melee_damage_modifier += rage_power  //Set rage melee damage bonus
```

Comment:
a single bonus point at max?

Reply:
This seems insignificant enough to skip.

Reply:
It's a multiplicative modifier

Reply:
To be clear, it's added to one, which then multiplies melee damage by that amount.